jetty-client/src/test/java/org/eclipse/jetty/client/HttpClientURITest.java

```diff
@@ -63,6 +67,51 @@ public void testIPv6Host() throws Exception
         Assert.assertEquals(HttpStatus.OK_200, request.send().getStatus());
     }
 
+    @Test(expected = IllegalArgumentException.class)
+    public void testIDNHost() throws Exception
+    {
+        startClient();
+        client.newRequest("http://пример.рф"); // example.com-like host in IDN domain
+    }
+
+    @Test(expected = IllegalArgumentException.class)
+    public void testIDNRedirect() throws Exception
+    {
+        String exampleHost = "http://пример.рф"; // example.com-like host in IDN domain
+        String incorrectlyDecoded = new String(exampleHost.getBytes(StandardCharsets.UTF_8), StandardCharsets.ISO_8859_1);
+
+        IncorrectRedirectServer server = new IncorrectRedirectServer(exampleHost);
```

Comment:
Rename the class `IDNRedirectServer` and make it a private static inner class of this test class. In this way, you don't need to worry about the license header. `IDNRedirectServer` is very specific to this test, so little reusable, so not worth to be a top level class.